pilot/pkg/config/memory/store.go

```diff
@@ -194,56 +193,25 @@ func (cr *store) Update(cfg config.Config) (string, error) {
 		return "", errNotFound
 	}
 
-	_, exists = ns.Load(cfg.Name)
+	oldCfg, exists := ns.Load(cfg.Name)
 	if !exists {
 		return "", errNotFound
 	}
 
-	existing, exists := ns.Load(cfg.Name)
-	if !exists {
-		return "", errNotFound
+	if cfg.ResourceVersion == oldCfg.(config.Config).ResourceVersion {
+		return cfg.ResourceVersion, nil
 	}
-	if hasConflict(existing.(config.Config), cfg) {
```

Comment:
We still need the conflict check. See https://github.com/istio/istio/commit/b53449263b1b23d68a04ddef64a9b6fe64946c6d for motivation.

The store is meant to act like Kubernetes. Updating with the wrong resource version is not legal, and by removing this check we make our tests which mock Kubernetes with memory store invalid.

Reply:
>Updating with the wrong resource version is not legal

This violates the client store even k8s client store does not work like this. Think about this scenario, client(mcp sink) only gets notified when resources changed, for virtualService, DestinationRule, ect, the resource versions are inherited from k8s, and every update means resource version change. It is right to update the store only when resource version differs or unset.


Haven't look into https://github.com/istio/istio/commit/b53449263b1b23d68a04ddef64a9b6fe64946c6d. i guess there is a different case.

Reply:
Agree with you @hzxuzhonghu 

@howardjohn  If we must use conflict check, only update cfg when the resource version is the same or not set, I can use other methods to fix this bug

Reply:
> This violates the client store even k8s client store does not work like this

My understanding is you need to set resourceVersion=0 or resourceVersion=oldCfg.resourceVersion in k8s or update will be rejected with "conflict error"

I see.. I guess there are two use cases for the store:
* Simulate k8s store, with exact semantics of k8s. This needs conflict check, as well as auto-generate creation timestamp, etc
* Use as a cache for another store, where we want to submit items as-is and have no modifications

So there is some tension here between these two use cases

Reply:
I think for a mcp server, the store will cache configs from k8s or other registries, we can update the cache regardless of the existing cache. In k8s，update means resource version must change. But i have to remind you that k8s is an eventual consistent model, we may not get all the update events(if the watch cache full, older events will be first dropped). 

For a MCP sink, the store will cache configs from a MCP server. Whether to update cache, there is two cases:

1. if a resource has resource version, update it when rv changes. So we can reduce many event handling if mcp server does not do delta push.
2. If a resource has no resource version, it is possible, because mcp protocol does not enforce it. In this case, we can not do delta handling, so we should update it in case miss any event.